File: archetypes/lambda/node24-typescript/src/handler.ts
import type { APIGatewayProxyHandlerV2 } from 'aws-lambda';
import { MongoClient } from 'mongodb';
import { Logger } from '@aws-lambda-powertools/logger';
import { z } from 'zod';

const logger = new Logger({ serviceName: process.env.SERVICE_NAME ?? 'mongodb-query' });
const schema = z.object({ id: z.string().min(1) });
let client: MongoClient | undefined;

async function db() {
  if (!client) {
    client = new MongoClient(process.env.MONGODB_URI!, {
      maxPoolSize: Number(process.env.MONGODB_MAX_POOL_SIZE ?? 10),
      connectTimeoutMS: Number(process.env.MONGODB_CONNECT_TIMEOUT_MS ?? 5000),
      serverSelectionTimeoutMS: Number(process.env.MONGODB_SERVER_SELECTION_TIMEOUT_MS ?? 5000)
    });
    await client.connect();
  }
  return client.db(process.env.MONGODB_DATABASE);
}

export const handler: APIGatewayProxyHandlerV2 = async (event) => {
  const parsed = schema.safeParse(event.pathParameters ?? {});
  if (!parsed.success) return { statusCode: 400, body: JSON.stringify({ code: 'INVALID_REQUEST' }) };

  const database = await db();
  const document = await database.collection(process.env.MONGODB_COLLECTION!).findOne(
    { businessId: parsed.data.id },
    { maxTimeMS: Number(process.env.MONGODB_MAX_TIME_MS ?? 3000) }
  );

  logger.info('MongoDB query completed', { found: Boolean(document) });
  return { statusCode: document ? 200 : 404, body: JSON.stringify({ data: document ?? null }) };
};

File: archetypes/lambda/python313/handler.py
import json
import os
from pymongo import MongoClient

_client = None

def get_collection():
    global _client
    if _client is None:
        _client = MongoClient(
            os.environ["MONGODB_URI"],
            maxPoolSize=int(os.getenv("MONGODB_MAX_POOL_SIZE", "10")),
            connectTimeoutMS=int(os.getenv("MONGODB_CONNECT_TIMEOUT_MS", "5000")),
            serverSelectionTimeoutMS=int(os.getenv("MONGODB_SERVER_SELECTION_TIMEOUT_MS", "5000")),
        )
    db = _client[os.environ["MONGODB_DATABASE"]]
    return db[os.environ["MONGODB_COLLECTION"]]

def handler(event, context):
    business_id = (event.get("pathParameters") or {}).get("id")
    if not business_id:
        return {"statusCode": 400, "body": json.dumps({"code": "INVALID_REQUEST"})}

    document = get_collection().find_one(
        {"businessId": business_id},
        max_time_ms=int(os.getenv("MONGODB_MAX_TIME_MS", "3000")),
    )

    return {
        "statusCode": 200 if document else 404,
        "body": json.dumps({"data": document}, default=str),
    }
